### DISCUSSION TITLE
Volatile Fulcurium Rods (Radiation should be 1 hull AOE and if gap open reduced effect on next hull)

### DISCUSSION THREAD

**Discussion by JoneKone** [FakeFishGames/Barotrauma#9392] [Ideas and suggestions]
I have been looking in to how radiation is implemented in the game, and while jovian radiation seems to be very penetrative and seriously it's fine.

~~Point radiation shouldn't have just an area of effect, I suggest (read feature request) that radiation would be more like radiation and the first step could be to make it act like a shadow casting light, the accuracy doesn't need to be so precise so i expect it could be less intense for the cpu. It would stop at all walls.~~

Edit. new idea, as .. nobody like's ray casting although I think it's fine xD (In fact I feel it's 100% fine, who's going to keep more than 1 fulcurium rod in every part of the ship) Also if no player in area there is no need to shadow cast the effect.)

```
The cheap computational version
1+x volatile fulcurium rods occupy 1 hull, this hull has 100% radiation effect. 
If gap open to next hull 25% radiation effect from set rods, on this hull. 

Making 1 volatile fulcurium rods effect maximum AOE to 5 possible hulls. 
```

(this is just so nobody else can suggest this)Next step **could be** to make it infect areas (read walls) and you can then scrub it with the sprayer and a ... special solution.

~~Making it more infectious would bring a whole other game play aspect~~ and I'm just **covering** this cause I want to, _not because I want it in the game._

To add decontamination showers that will clean when powered the toxic waste of the suit. (Possibly the green glow of the suit would warn you off)

**Comment by Oukido** [comment]
If there was not such a high performance cost in all the required calculations I would love that idea. But since the game is already suffering I would advice against such change. Imagine a storage container case full with stacks of volatile rods doing its thing.

**Reply by JoneKone** [reply]
Hadn't thought of that.. then again, ... ... it could check.. we have 1 + extra volatile fulcurium rods in 1 hull, lets radiate that area and hey there is a gap open lets radiate that area.. so in a sense no tracing needed? (no shadow casting needed)

**Comment by JoneKone** [comment]
![suggested_radiation_aoe](https://user-images.githubusercontent.com/1114115/174750985-3e0747b0-f4fa-4be4-b5be-e90b152261f9.png)

Forgot to add the gaps, but I'm sure everybody can imagine the gaps. The gaps are open in this..

**Reply by JoneKone** [reply]
Tho whom it may concern, if they is reading this. With this hull based model I'm sure the AI can avoid these areas.. because we know where the radiation is.. .. easily.
